### PR TITLE
integration Tests actions template

### PR DESCRIPTION
**Pull request by rangan-anand**
## What is the goal of this PR?

we are making changes to add in integration Tests actions template to use in other services. to create a seperate stack then run the process and terminate the stack

### CODE REVIEW

**Review by thomaschristopherking — APPROVED** [review]

File: .github/workflows/build_and_deploy_integration_test_template.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        run: sam build --template-file ${{ inputs.TEMPLATE_NAME }} --use-container -p --container-env-var GITHUB_TOKEN=$GITHUB_TOKEN
+      - name: Deploy SAM template
+        run: sam deploy --config-file ${{ inputs.TOML_CONFIG_FILE }} --config-env ${{ inputs.TOML_CONFIG }} --capabilities CAPABILITY_NAMED_IAM --no-confirm-changeset --no-fail-on-empty-changeset --stack-name ${{ inputs.SAM_STACK_NAME }}
+      - name: Get Deployed Step Function Details
+        run: |
+          {
+            echo 'STACK_INFO<<EOF'
+            (sam list stack-outputs --stack-name core-data-pipeline-integration-test --output json)
+            echo EOF
+          } >> "$GITHUB_ENV"
```
Comment (ukyen8) [inline .github/workflows/build_and_deploy_integration_test_template.yaml:65]: Where will this be used?

File: .github/workflows/build_and_deploy_integration_test_template.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          if [ -f requirements_testing.txt ]; then pip install -r requirements_testing.txt; fi
+      - name: Run Integration Test
+        run: |
+          python -m integration_tests
```
Comment (ukyen8) [inline .github/workflows/build_and_deploy_integration_test_template.yaml:73]: Is it `python -m pytest integration_tests`?

File: .github/workflows/build_and_deploy_integration_test_template.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Delete SAM template
```
Comment (ukyen8) [inline .github/workflows/build_and_deploy_integration_test_template.yaml:74]: Why do we need to delete the stack?

**Review by ukyen8 — COMMENTED** [review]
Some questions

File: .github/workflows/build_and_deploy_integration_test_template.yaml (hunk 1 above)
Comment (rangan-anand) [reply .github/workflows/build_and_deploy_integration_test_template.yaml:65]: this is to set STACK_INFO as environmental variable, this exports all the outputs/ Arns which can be used in the integrations tests ( line 71)

File: .github/workflows/build_and_deploy_integration_test_template.yaml (hunk 2 above)
Comment (rangan-anand) [reply .github/workflows/build_and_deploy_integration_test_template.yaml:73]: we are planning to do integration test (invoke the step function directly from github actions) , so this needs to run the script in integration_tests folder

File: .github/workflows/build_and_deploy_integration_test_template.yaml (hunk 3 above)
Comment (rangan-anand) [reply .github/workflows/build_and_deploy_integration_test_template.yaml:74]: as we are deploying a new stack just for integration testing, this step is to clean up the stack once the run completes

Comment (ukyen8) [reply .github/workflows/build_and_deploy_integration_test_template.yaml:74]: okay, I see, makes sense!

**Review by ukyen8 — APPROVED** [review]

**Pull request merged**